Fixing SonarLint Errors

burkodigergely opened the pull request:
I ran SonarLint on the project and fixed the errors thrown by the program

### Description

<!-- Please include a summary of the change and which issue is fixed (i.e. `closes #XXXX`). -->

### Checklist

- [x] An issue was first created **before** opening this pull request
- [x] The new code follows the [contribution guidelines](CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests to ensure that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

burkodigergely committed:
Fixing SonarLint Errors

I ran SonarLint on the project and fixed the errors thrown by the program

AkosUzonyi commented on DataStructures/Dictionaries/CuckooHashTable.cs:
```diff
@@ -114,7 +114,7 @@ private void _rehash()
         private void _rehash(int newCapacity)
         {
-            int primeCapacity = PRIMES.GetNextPrime(newCapacity);
+            _ = PRIMES.GetNextPrime(newCapacity);
```
Maybe we could even remove that underline too

AkosUzonyi replied:
Looking more closely at this, looks like this line and the PRIMES field can be removed, they are not used at all.

AkosUzonyi commented on DataStructures/Dictionaries/CuckooHashTable.cs:
```diff
@@ -164,9 +164,9 @@ private int _cuckooHash(TKey key, int whichHashFunction)
         {
             if (index < 0 || index > _collection.Length)
-                throw new IndexOutOfRangeException();
+                throw new ArgumentException($"Invalid argument");
```
I'm not sure that ArgumentException is beter than IndexOutOfRangeException here.

AkosUzonyi commented on DataStructures/Graphs/UndirectedWeightedDenseGraph.cs:
```diff
@@ -109,10 +109,10 @@ public virtual IEnumerable<WeightedEdge<T>> Edges
         {
             if (!HasVertex(vertex))
-                throw new ArgumentOutOfRangeException("One of vertex is not part of the graph.");
+                throw new ArgumentOutOfRangeException($"One of vertex is not part of the graph.");
```
I think we shouldn't add a "$" before strings if we don't use string interpolation.
But it would be a good idea to use interpolation for printing the vertex.

AkosUzonyi replied:
So I suggest to print the actual vertex here with interpolation.

And in the other places where "$" is inserted, either use interpolation actually, or if there is no variable to print, just remove the "$" sign.

AkosUzonyi commented on DataStructures/Hashing/PrimeHashingFamily.cs:
```diff
@@ -68,11 +68,11 @@ public void GenerateNewFunctions()
             int preHashValue = 0;
             int multiplier = _multipliersVector[whichHashFunction - 1];
-            var characters = preHashedKey.ToString().ToCharArray();
+            _ = preHashedKey.ToString().ToCharArray();
```
I think this line could be removed entirely

AkosUzonyi commented on DataStructures/Heaps/BinomialMinHeap.cs:
```diff
@@ -113,6 +116,13 @@ private void _removeAtIndex(int minIndex)
 
+        private void CheckCapacity(int capacity)
+        {
+            string message = $"Capacity is negative: {capacity}";
+            if (capacity <= 0)
+                throw new ArgumentOutOfRangeException(message);
```
This message variable could be inlined in the exception constructor.

AkosUzonyi reviewed (changes requested):
Nice :)

Please also add a new markdown documentation file about static analysis (eg. `doc/static_analysis.md`) containing a short description of these changes.

AkosUzonyi commented:
I really look forward to an update on this pull request.

burkodigergely committed:
Removing PRIMES field

the PRIMES field was not used anywhere so we thought it would be better to remove

burkodigergely committed:
Change  to IndexOutOfRangeException

Change  back to IndexOutOfRangeException at line 167

burkodigergely committed:
Fixing requred  format bug

Fixing the requested format bug with the string given in the exception at line 115

MarcellHarmaci commented on DataStructures/Heaps/BinomialMinHeap.cs:
```diff
@@ -182,13 +192,14 @@ public bool IsEmpty
         {
             if (newCollection == null)
-                throw new ArgumentNullException();
+                throw new ArgumentNullException($"The provided parameter was null");
```
Why are we using parameterizable strings when there are no parameters?
This question also applies to the other exception messages without parameters.

gyupetosz replied:
SonarLint suggested to extend the exception descreption so we made it open for future clarifications by added variables.

AkosUzonyi replied:
I agree with @MarcellHarmaci, the "$" sign is unnecessary here.

The added message could be useful, but then the parameter name should be included.

burkodigergely committed:
Removing unnesesary line

Removing the unnesesary line at row 75: _ = preHashedKey.ToString().ToCharArray() It was not used

burkodigergely committed:
Make code more readable

Make  the code more readable by  message variable be inlined in the exception constructor at line 122
